main/src/cgeo/geocaching/utils/BackupUtils.java

```diff
@@ -65,69 +70,69 @@ public BackupUtils(final Activity activityContext) {
+        contentStorageHelper.selectFolder(PersistableFolder.BACKUP.getUri(), this::restore);
+    }
+
     /**
-     * Public methods also containing question dialogs, etc
+     * Show restore dialog
      */
-
     @SuppressLint("SetTextI18n")
-    public void restore(final File backupDir) {
-        if (!hasBackup(backupDir)) {
-            Toast.makeText(activityContext, R.string.init_backup_no_backup_available, Toast.LENGTH_LONG).show();
-        } else {
-            // We are using ContextThemeWrapper to prevent crashes caused by missing attribute definitions when starting the dialog from MainActivity
-            final Context c = new ContextThemeWrapper(activityContext, Settings.isLightSkin() ? R.style.Dialog_Alert_light : R.style.Dialog_Alert);
-            final View content = LayoutInflater.from(c).inflate(R.layout.restore_dialog, null);
-            final CheckBox databaseCheckbox = content.findViewById(R.id.database_check_box);
-            final CheckBox settingsCheckbox = content.findViewById(R.id.settings_check_box);
-            final TextView warningText = content.findViewById(R.id.warning);
-
-            if (getDatabaseBackupTime(backupDir) != 0) {
-                databaseCheckbox.setText(activityContext.getString(R.string.init_backup_caches) + "\n(" + Formatter.formatShortDateTime(getDatabaseBackupTime(backupDir)) + ")");
-                databaseCheckbox.setEnabled(true);
-                databaseCheckbox.setChecked(true);
-            } else {
-                databaseCheckbox.setText(activityContext.getString(R.string.init_backup_caches) + "\n(" + activityContext.getString(R.string.init_backup_unavailable) + ")");
-            }
-            if (getSettingsBackupTime(backupDir) != 0) {
-                settingsCheckbox.setText(activityContext.getString(R.string.init_backup_program_settings) + "\n(" + Formatter.formatShortDateTime(getSettingsBackupTime(backupDir)) + ")");
-                settingsCheckbox.setEnabled(true);
-                settingsCheckbox.setChecked(true);
-            } else {
-                settingsCheckbox.setText(activityContext.getString(R.string.init_backup_program_settings) + "\n(" + activityContext.getString(R.string.init_backup_unavailable) + ")");
-            }
+    public void restore(final Folder backupDir) {
 
-            final AlertDialog dialog = Dialogs.newBuilder(activityContext)
-                    .setTitle(activityContext.getString(R.string.init_backup_restore))
-                    .setView(content)
-                    .setPositiveButton(activityContext.getString(android.R.string.yes), (alertDialog, id) -> {
-                        final boolean database = databaseCheckbox.isChecked();
-                        final boolean settings = settingsCheckbox.isChecked();
+        if (backupDir == null) {
+            return;
+        }
 
-                        alertDialog.dismiss();
+        if (!hasBackup(backupDir)) {
```

Comment:
This one is called afterwards. `hasBackup` should return true, but it isn't...

`backupDir` looks normal at a first glance (correct URI)